### PR TITLE
fix: Get the ValueType from db instead of RHS

### PR DESCRIPTION
**Pull request by sanket-mundra**
## Description

Get the ValueType from db instead of rhs value

### CODE REVIEW

File: .github/workflows/pr-test.yml [hunk 1]
```diff
@@ -54,6 +54,7 @@ jobs:
         uses: hypertrace/github-actions/gradle@main
         with: 
           args: copyAllReports --output-dir=/tmp/test-reports
+        if: always()
```
Comment (suresh-prakash) [inline .github/workflows/pr-test.yml:57]: Why is this change required?

Comment (suresh-prakash) [reply .github/workflows/pr-test.yml:57]: Can you confirm with @shaishaw if that is not an issue?

Comment (shaishaw) [reply .github/workflows/pr-test.yml:57]: Is the intention is to execute the step "copy test reports" on all case ? then it's fine to keep always flag.

Comment (sanket-mundra) [reply .github/workflows/pr-test.yml:57]: What should be the ideal scenario? We want to see the result of integration test only in case of failure? Also, this is how it was in other repos as well so, I did that: https://github.com/hypertrace/attribute-service/blob/main/.github/workflows/pr-test.yml#L40

Comment (shaishaw) [reply .github/workflows/pr-test.yml:57]: If you want it to run only incase of failure, you can try with other options.

you can try with if: failure()

Comment (aaron-steinfeld) [reply .github/workflows/pr-test.yml]: We want test results whether success or failure of the tests. More accurate would be to look for the test task to succeed or fail (e.g. `success() || failure()`) to avoid upload in cases where the task isn't reached/cancelled, but we usually just use always() as a shorthand since it's more readable and even in the cancel case a failed upload doesn't really matter.

Comment (sanket-mundra) [reply .github/workflows/pr-test.yml]: Updated it to `if: failure()`

Comment (sanket-mundra) [reply .github/workflows/pr-test.yml]: We will need to update the same for `Archive test reports` job as well as the report will be copied only in case of failure and only then we need to archive it:

Comment (aaron-steinfeld) [reply .github/workflows/pr-test.yml]: We typically upload on success too - you can use the other repos you were looking at for reference.

Comment (sanket-mundra) [reply .github/workflows/pr-test.yml]: I other repos, we are simply using `if: always()`. So, reverting back to `if: always()`

**Review by suresh-prakash — APPROVED** [review]
